man/man8/zfs.8

```diff
@@ -983,6 +1009,35 @@ This property can also be referred to by its shortened column name,
 
 .sp
 .ne 2
+.na
+\fB\fBencryption\fR=\fBoff\fR | \fBon\fR | \fBaes-128-ccm\fR | \fBaes-192-ccm\fR | \fBaes-256-ccm\fR | \fBaes-128-gcm\fR | \fBaes-192-gcm\fR | \fBaes-256-gcm\fR
+.ad
+.sp .6
+.RS 4n
+Controls the encryption cipher suite (block cipher, key length, and mode) used for this dataset. Requires the encryption feature to be enabled on the pool. Requires a keysource to be set at dataset creation time.
+.sp
+Selecting \fBencryption\fR=\fBon\fR when creating a dataset indicates that the current default encryption suite will be selected, which is currently \fBaes-256-ccm\fR. In order to provide consistent data protection, encryption must be specified at dataset creation time and it cannot be changed afterwards.
+.sp
+The \fBencryption\fR property is inherited by child datasets by default. All inheriting datasets will be accessible when the root key is mounted. A child dataset may also set encryption locally, which will indicate that it should have a separate key and \fBkeysource\fR from the parent. Regardless, all children of an encrypted dataset must also be encrypted.
+.sp
+For more details and caveats about encryption see \fBzpool-features\fR.
+.RE
+
+.sp
+.ne 2
+.mk
+.na
+\fB\fBkeysource\fR=<\fBraw\fR | \fBhex\fR | \fBpassphrase\fR>,<\fBprompt\fR | \fBfile://\fR\fI<absolute file path\fR>>
+.ad
+.sp .6
+.RS 4n
+Controls what format the user's encryption key will be provided as and where it should be loaded from. This property is only set when the dataset is encrypted.
+.sp
+Raw keys and hex keys must be 32 bytes long (regardless of the chosen encryption suite) and must be randomly generated. Passphrases must be between 8 and 64 bytes long and will be processed through PBKDF2 before being used (see the \fBpbkdf2iters\fR property). Even though the encryption suite cannot be changed after dataset creation, the keysource can be with \fBzfs key -c\fR. If prompt is selected as the key location ZFS will ask for the key at the command prompt when it is required (\fBzfs mount\fR and \fBzfs key -l\fR). If \fBfile\fR is selected, the key will be loaded from the specified absolute file path. The key may also be passed in via STDIN, but users should be careful not to place keys which should be kept secret on the command line.
```

Comment:
That is the point I was trying to make. I decided to put it here because otherwise I would have to replicate the same text for `zfs mount`, `zfs key -l` and `zpool import -l`